app/api_topology.go

```diff
@@ -30,10 +30,19 @@ type APIEdge struct {
 	Metadata report.AggregateMetadata `json:"metadata"`
 }
 
+func render(rpt report.Report, maps []topologyMapper) report.RenderableNodes {
+	result := report.RenderableNodes{}
+	for _, m := range maps {
+		rns := m.selector(rpt).RenderBy(m.mapper, m.pseudo)
+		result.Merge(rns)
+	}
+	return result
+}
```

Comment:
Yes the rest is plumbing.

Going forward, once we can chain these mapping functions, you would make this recursive.